### PR TITLE
Tt 102 ros mmitss component testing

### PR DESCRIPTION
**Pull request by paulbourelly999**
# PR Details
## Description
This PR includes new documentation to setup the related environment as an integration testing environment for testing this MMITSS Ros2 Transceiver

## Related Jira Key

[TT-102](https://usdot-carma.atlassian.net/browse/TT-102)

## How Has This Been Tested?
Since there are bugs indentified for mmitss transciever that the CarmaDecoderRos2 doesnot launch ros node, only isolated local testing has been conducted to ensure the mmitss transciever is up and running.  

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

[TT-102]: https://usdot-carma.atlassian.net/browse/TT-102?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/vsp/testSRM/__pycache__/test_SRM.cpython-311-pytest-7.4.0.pyc
Comment (paulbourelly999) [inline test/vsp/testSRM/__pycache__/test_SRM.cpython-311-pytest-7.4.0.pyc:1]: This file does not need to be committed. It is python compiled byte code that should ideally be generated when someone attempts to run the script for the first time.

File: test/vsp/testBsm/runTest.sh [hunk 1]
```diff
@@ -1,6 +1,8 @@
...
 
 source /opt/ros/foxy/setup.bash
-source ../../../other/install/setup.bash
+#source ../../../other/install/setup.bash
+source /home/yilin/SIP_project/carma-msgs/install/setup.bash 
+source ../../../mmitss/src/common/MsgTransceiver/CarmaDecoderRos2/install/setup.bash
 
-../../../mmitss/src/common/MsgTransceiver/CarmaDecoderRos2/src/M_CarmaMsgDecoder & sleep 2;pytest test_hostBsm.py
\ No newline at end of file
+../../../mmitss/src/common/MsgTransceiver/CarmaDecoderRos2/src/M_CarmaMsgDecoder & sleep 2;python3 -m test_hostBsm.py
```
Comment (paulbourelly999) [inline test/vsp/testBsm/runTest.sh]: There is a way to run pytest similar to this. Apparently you can run something like
```bash
python3 -m unittest test_hostBsm.py
```
(https://docs.python.org/3/library/unittest.html)
Have you tried this?

File: test/vsp/testBsm/test_hostBsm.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+ #!/usr/bin/env python3
```
Comment (paulbourelly999) [inline test/vsp/testBsm/test_hostBsm.py]: Did you add this shebang on purpose? This should allow you to just call this as a script.
```
./test_hostBsm.py
```

Comment (Dylan-Wyl10) [reply test/vsp/testBsm/test_hostBsm.py]: I used  this to make sure the script is running by python3. I can remove this if we running with a script.

File: test/vsp/testSRM/__pycache__/test_SRM.cpython-311-pytest-7.4.0.pyc
Comment (Dylan-Wyl10) [reply test/vsp/testSRM/__pycache__/test_SRM.cpython-311-pytest-7.4.0.pyc:1]: i will remove this.

File: test/vsp/testBsm/test_hostBsm.py (hunk 2 above)
Comment (paulbourelly999) [reply test/vsp/testBsm/test_hostBsm.py]: I think this article maybe helpful (https://realpython.com/python-shebang/). Specifically it states "In short, a [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)) is a special kind of [comment](https://realpython.com/python-comments-guide/) that you may include in your source code to tell the operating system’s [shell](https://en.wikipedia.org/wiki/Shell_(computing)) where to find the interpreter for the rest of the file"
"A shebang is only relevant to [runnable scripts](https://realpython.com/run-python-scripts/) that you wish to execute without explicitly specifying the program to run them through."
Basically having the shebang and then calling the script `python3 <script_name>` defeats the purpose of the shebang.

File: test/vsp/testBsm/runTest.sh (hunk 1 above)
Comment (Dylan-Wyl10) [reply test/vsp/testBsm/runTest.sh]: Yes I have tries this. I think it is better to keep the original script in order to close this story. Then I can work on debuging the ros2 issue.

File: test/vsp/testBsm/runTest.sh [hunk 3]
```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 
 source /opt/ros/foxy/setup.bash
-source ../../../other/install/setup.bash
+#source ../../../other/install/setup.bash
+source /home/yilin/SIP_project/carma-msgs/install/setup.bash
```
Comment (paulbourelly999) [inline test/vsp/testBsm/runTest.sh]: Lets remove this. I think as long as you source the built transceiver bash you should be ok. We want to refrain from using paths that would break in someone else's environment

**Review by Dylan-Wyl10 — APPROVED** [review]
address all the comment and ready to merge
